libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/actionbutton/RecenterButtonBinder.kt

```diff
@@ -0,0 +1,30 @@
+package com.mapbox.navigation.dropin.actionbutton
+
+import android.view.ViewGroup
+import androidx.lifecycle.viewModelScope
+import com.mapbox.navigation.base.ExperimentalPreviewMapboxNavigationAPI
+import com.mapbox.navigation.core.lifecycle.MapboxNavigationObserver
+import com.mapbox.navigation.dropin.camera.RecenterButtonComponentContractImpl
+import com.mapbox.navigation.dropin.internal.extensions.reloadOnChange
+import com.mapbox.navigation.dropin.navigationview.NavigationViewContext
+import com.mapbox.navigation.ui.base.lifecycle.UIBinder
+import com.mapbox.navigation.ui.base.view.MapboxExtendableButton
+import com.mapbox.navigation.ui.maps.internal.ui.RecenterButtonComponent
+
+@ExperimentalPreviewMapboxNavigationAPI
+internal class RecenterButtonBinder(
+    private val context: NavigationViewContext
+) : UIBinder {
+
+    override fun bind(viewGroup: ViewGroup): MapboxNavigationObserver {
+        return reloadOnChange(context.styles.recenterButtonParams) { params ->
+            val button = MapboxExtendableButton(viewGroup.context, null, 0, params.style)
+            viewGroup.removeAllViews()
+            viewGroup.addView(button, params.layoutParams)
+
+            RecenterButtonComponent(button) {
+                RecenterButtonComponentContractImpl(context.viewModel.viewModelScope, context.store)
+            }
+        }
```

Comment:
Why not do the same as you are doing here https://github.com/mapbox/mapbox-navigation-android/pull/6490/files#diff-bfe82a9aa5902eb8b53e4b16729122d2707e7ee830fc65c37ea592410fcc09f1R16-R21

Reply:
Answered here https://github.com/mapbox/mapbox-navigation-android/pull/6490#discussion_r999887338